### PR TITLE
refactored project with adding service layer

### PR DESCRIPTION
**Pull request by vcaf**

### CODE REVIEW

File: src/Feedback-Service/Dtos/CreateFeedbackDto.cs [hunk 1]
```diff
@@ -1,8 +1,25 @@
...
+
+// namespace Feedback_Service.Dtos;
+
+// public record CreateFeedbackDto
+// (
+//     [Required] int PatientId,
+//     [Required] Guid AuthorId,
+//     [Optional] Guid? StressMeassurementId,
+//     [Optional] string? FeedbackComment
+// );
```
Comment (milankoster) [inline src/Feedback-Service/Dtos/CreateFeedbackDto.cs]: Is there a way we can keep this as a CreateFeedbackDto record?

File: src/Feedback-Service/Dtos/CreateFeedbackDto.cs [hunk 2]
```diff
@@ -1,8 +1,25 @@
...
+
 namespace Feedback_Service.Dtos;
 
-public class CreateFeedbackDTO
+public class CreateFeedbackDto
 {
-    public Guid PatientId { get; set; }
+    public int PatientId { get; set; }
+    public Guid AuthorId { get; set; }
     public Guid StressMeassurementId { get; set; }
```
Comment (milankoster) [inline src/Feedback-Service/Dtos/CreateFeedbackDto.cs:10]: I believe you made a small typo here: `StressMeassurementId` should be `StressMeassurementId`

File: src/Feedback-Service/Dtos/UpdateFeedbackDto.cs [hunk 3]
```diff
@@ -1,8 +1,9 @@
...
 
-public class UpdateFeedbackDTO
+public class UpdateFeedbackDto
 {
-    public Guid PatientId { get; set; }
+    public int PatientId { get; set; }
+    public Guid AuthorId { get; set; }
     public Guid StressMeassurementId { get; set; }
-    public string Feedback { get; set; } = "";
+    public string? FeedbackComment { get; set; }
```
Comment (milankoster) [inline src/Feedback-Service/Dtos/UpdateFeedbackDto.cs:8]: I believe this too could be a record. If that causes any issues, let's have a look together.

File: src/Feedback-Service/Models/Feedback.cs [hunk 4]
```diff
@@ -1,11 +1,19 @@
...
+
 namespace Feedback_Service.Models;
 
 public class Feedback
 {
     public Guid Id { get; set; }
-    public Guid PatientId { get; set; }
+    public int PatientId { get; set; }
     public Guid AuthorId { get; set; }
     public Guid StressMeassurementId { get; set; }
```
Comment (milankoster) [inline src/Feedback-Service/Models/Feedback.cs:10]: I believe this is also intended to be `StressMeasurementId`.

File: src/Feedback-Service/Services/FeedbackService.cs [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+        return feedbackEntry.AsDto();
+    }
+
+    public async Task<IEnumerable<FeedbackDto>> GetPatientFeedbackEntryById(int patientId)
+    {
+        var feedbackEntries = (await feedbackEntriesRepository.GetPatientFeedbacks(patientId))
+                            .Select(feedbackEntry => feedbackEntry.AsDto());
+
+        System.Diagnostics.Debug.WriteLine(feedbackEntries);
```
Comment (milankoster) [inline src/Feedback-Service/Services/FeedbackService.cs:40]: I think this was probably used to debugging. Interesting that you used `System.Diagnostics.Debug`, I have never seen it before, but it's really cool! Apparently it is removed by the compiler on runtime and is a lot faster than `Console.Writeline`, so I'll definitely be stealing this 😉

File: src/Feedback-Service/Services/FeedbackService.cs [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        {
+            PatientId = createFeedbackDto.PatientId,
+            AuthorId = createFeedbackDto.AuthorId,
+            StressMeassurementId = createFeedbackDto.StressMeassurementId,
+            FeedbackComment = createFeedbackDto.FeedbackComment,
+            CreatedDate = DateTimeOffset.UtcNow
+        };
+
+        await feedbackEntriesRepository.Create(feedbackEntry);
+        return feedbackEntry.AsDto();
```
Comment (milankoster) [inline src/Feedback-Service/Services/FeedbackService.cs:56]: You could consider adding the empty line between await and return for consistency, but it's obviously nothing code-breaking.

```suggestion
        await feedbackEntriesRepository.Create(feedbackEntry);
        
        return feedbackEntry.AsDto();
```

**Review by milankoster — CHANGES_REQUESTED** [review]

**Comment by milankoster** [comment]
Forgot to add this before submitting, but really nice job refactoring! The service layer will be really helpful when we start publishing the feedback entries. And the separation in DTOs is really nice for consistency sake with the other microservices. Overall some really beneficial changes that will benefit the code quality a lot in the long run 😄

File: src/Feedback-Service/Services/FeedbackService.cs (hunk 5 above)
Comment (vcaf) [reply src/Feedback-Service/Services/FeedbackService.cs:40]: haha thank you and i totally forgot to remove it.

File: src/Feedback-Service/Dtos/CreateFeedbackDto.cs (hunk 1 above)
Comment (vcaf) [reply src/Feedback-Service/Dtos/CreateFeedbackDto.cs]: yesss, just resolved it

File: src/Feedback-Service/Dtos/UpdateFeedbackDto.cs (hunk 3 above)
Comment (vcaf) [reply src/Feedback-Service/Dtos/UpdateFeedbackDto.cs]: yesss, just resolved it

**Pull request closed without merging**